test/DrawAndPrizeDistributionTimelock.test.ts

```diff
@@ -0,0 +1,98 @@
+
+const { getSigners } = ethers;
+
+describe('DrawAndPrizeDistributionTimelock', () => {
+  let wallet1: any;
+  let wallet2: any;
+
+  let drawAndPrizeDistributionTimelock: Contract;
+
+  let prizeDistributionFactory: MockContract;
+  let drawCalculatorTimelock: MockContract;
+  let drawBuffer: MockContract;
+
+  let drawAndPrizeDistributionTimelockFactory: ContractFactory;
+
+  beforeEach(async () => {
+    [wallet1, wallet2] = await getSigners();
+
+    const PrizeDistributionFactory = await artifacts.readArtifact('IPrizeDistributionFactory');
+    prizeDistributionFactory = await deployMockContract(wallet1, PrizeDistributionFactory.abi);
+
+    const DrawBufferArtifact = await artifacts.readArtifact('IDrawBuffer');
+    drawBuffer = await deployMockContract(wallet1, DrawBufferArtifact.abi);
+
+    const DrawCalculatorTimelock = await artifacts.readArtifact('DrawCalculatorTimelock');
+    drawCalculatorTimelock = await deployMockContract(wallet1, DrawCalculatorTimelock.abi);
+
+    drawAndPrizeDistributionTimelockFactory = await ethers.getContractFactory('DrawAndPrizeDistributionTimelock');
+
+    drawAndPrizeDistributionTimelock = await drawAndPrizeDistributionTimelockFactory.deploy(
+      wallet1.address,
+      drawBuffer.address,
+      prizeDistributionFactory.address,
+      drawCalculatorTimelock.address,
+    );
+  });
+
+  describe('constructor()', () => {
+    it('should emit Deployed event', async () => {
+      await expect(drawAndPrizeDistributionTimelock.deployTransaction)
+        .to.emit(drawAndPrizeDistributionTimelock, 'Deployed')
+        .withArgs(
+          drawBuffer.address,
+          prizeDistributionFactory.address,
+          drawCalculatorTimelock.address,
+        );
+
+      expect(await drawAndPrizeDistributionTimelock.drawBuffer()).to.equal(drawBuffer.address);
+
+      expect(await drawAndPrizeDistributionTimelock.prizeDistributionFactory()).to.equal(
+        prizeDistributionFactory.address,
+      );
+      expect(await drawAndPrizeDistributionTimelock.timelock()).to.equal(drawCalculatorTimelock.address);
+    });
+  });
+
+  describe('push()', () => {
+    const draw: any = {
+      drawId: ethers.BigNumber.from(0),
+      winningRandomNumber: ethers.BigNumber.from(1),
+      timestamp: ethers.BigNumber.from(10),
+      beaconPeriodStartedAt: Math.floor(new Date().getTime() / 1000),
+      beaconPeriodSeconds: 1000,
+    };
+
+    it('should allow a push when no push has happened', async () => {
+      await drawBuffer.mock.pushDraw.returns(draw.drawId);
+      await prizeDistributionFactory.mock.pushPrizeDistribution.returns();
+      await drawCalculatorTimelock.mock.lock.returns(true);
+      await expect(drawAndPrizeDistributionTimelock.push(draw, BigNumber.from(1000000)))
+        .to.emit(drawAndPrizeDistributionTimelock, 'DrawAndPrizeDistributionPushed');
+    });
+
+    it('should not allow a push from a non-owner', async () => {
```

Comment:
Also add a test case to be able to call the function with a manager.

Reply:
Does that test belong in this contract? Should the Owanable contract test that?